### ISSUE TITLE
Bug in `div` operation

### ISSUE THREAD

**Issue by polydez** [0xPolygonMiden/miden-vm#1186]
Hi there,
I've faced a bug in Miden assembly `div` operation. 

Please, look into this MASM code snippet:

```
begin
  # Works as expected, result is 5
  push.10
  div.2 

  # Unexpected result: 9223372034707292166
  push.11
  div.2

  # Works as expected, result is 6
  push.12
  div.2
  
  # Unexpected result: 9223372034707292167
  push.13
  div.2

  # Works as expected, result is 2
  push.8
  div.4


  # Unexpected result: 7378697627765833730
  push.8
  div.5

  # Works as expected, result is 5
  push.25
  div.5
end
```

It seems to work good when there is no fractional part in result. Otherwise it gives weird result. Maybe it's due to encoding of modulo and I need to apply additional math on result in order to extract only fixed part? I haven't found information on that matter in docs.

**Comment by bobbinth** [comment]
This is actually not a bug - `div` instruction performs division in the prime field, which is defined via inversion (i.e. $c \leftarrow a \cdot b^{-1}$ [see here](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/field_operations.html#arithmetic-and-boolean-operations)). We should probably document this better in the [Field Operations](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/field_operations.html#field-operations) section.

For integer division, we can use `u32div` or `u32divmod` instructions.

**Comment by bobbinth** [comment]
Closed by #1192.
